src/ast/codegen_llvm.cpp

```diff
@@ -1124,7 +1124,7 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
   }
   b_.CreateMapUpdateElem(map, key, val);
   b_.CreateLifetimeEnd(key);
-  if (!assignment.expr->is_variable)
+  if (!(assignment.expr->is_variable || assignment.expr->type.type == Type::inet))
```

Comment:
Why doesn't inet get a lifetime end? Actually, I wonder if variables can have lifetime end too now that they live on the stack

Reply:
I guess it doesn't. It was required before (I think in this case `$a = ntop(2,0); @a = $a; printf("%s\n", $a)`), but doesn't seem to be required now. As for variables, I couldn't come up with any scenario where ending their lifetime here would cause any issues, so I guess we can remove it (but I would suggest doing this in another PR, after we release 0.9.1).

Found another similar bug while debugging this: arrays were also not working properly with maps (`tracepoint:tcp:tcp_set_state { $a = args->saddr_v6; }`). I'll add a commit to fix it as well.

Reply:
Changed my mind, arrays inside maps won't work properly with just that. I'll drop that commit and open an issue.